wodles/aws/aws-s3.py

```diff
@@ -437,33 +455,36 @@ def remove_none_fields(event):
     def get_full_prefix(self, account_id, account_region):
         raise NotImplementedError
 
-    def build_s3_filter_args(self, aws_account_id, aws_region):
+    def build_s3_filter_args(self, aws_account_id, aws_region, iterating=False):
         filter_marker = ''
         if self.reparse:
             if self.only_logs_after:
-                filter_marker = self.marker_only_logs_after(aws_region, aws_account_id)
+                filter_marker = self.marker_only_logs_after(aws_region, aws_account_id, self.only_logs_after)
         else:
-            # Where did we end last run thru on this account/region?
-            query_results = self.db_connector.execute(AWSBucket.sql_find_last_log_processed.format(aws_account_id=aws_account_id,
-                                                                                         aws_region=aws_region))
+            query_last_key = self.db_connector.execute(self.sql_find_last_key_processed.format(table_name=self.db_table_name,
+                                                                                        aws_account_id=aws_account_id,
+                                                                                        aws_region=aws_region))
             try:
-                created_date = query_results.fetchone()[0]
-                # Existing logs processed, but older than only_logs_after
-                if created_date > int(self.only_logs_after.strftime('%Y%m%d')):
-                    self.only_logs_after = datetime.strptime(str(created_date), '%Y%m%d')
-                filter_marker = self.marker_only_logs_after(aws_region, aws_account_id)
+                last_key = query_last_key.fetchone()[0]
```

Comment:
I'd add a handler for `KeyError` exceptions too:
```python
except (TypeError, KeyError) as e:
```

Reply:
Fixed!